### PR TITLE
Add trending tags widget to community page

### LINKED ISSUE #7473: Trending hashtags feature
### Expected behavior
Hashtags are useful. But the discovery of interesting topics/hashtags isn't easy. Would be cool if we would have some kind of trending hashtags feature. See Twitter. :-)

I don't know if this is actually possible in a distributed network and I don't know if we have the necessary data in our database. It's just an idea I would like to see as a Friendica feature.

### Friendica version you encountered the "problem"
2019.09-dev

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #7473

This aside widget displays global or local trending tags depending on the current community page. Its display is governed by a new additional feature in the General tab. 10 results are displayed at all times, and a spoiler block hides 10 more if any.

Results from either global/local queries are cached for an hour to reduce performance hit.

### CODE REVIEW

File: src/Content/Widget/TrendingTags.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		return $o;
+	}
+
+	/**
+	 * Returns a list of the most frequent global tags over the given period
+	 *
+	 * @param int $period Period in hours to consider posts
+	 * @return array
+	 * @throws \Exception
+	 */
```
Comment (nupplaphil) [inline src/Content/Widget/TrendingTags.php]: This seems for me like a `Friendica\Model\Tags` class method. I don't think we should mix it with Content methods.

File: src/Content/Widget/TrendingTags.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	/**
+	 * Returns a list of the most frequent local tags over the given period
+	 *
+	 * @param int $period Period in hours to consider posts
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function getLocalTrendingTags(int $period)
```
Comment (nupplaphil) [inline src/Content/Widget/TrendingTags.php]: This seems for me like a `Friendica\Model\Tags` class method. I don't think we should mix it with Content methods.

File: src/Content/Widget/TrendingTags.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+	/**
+	 * Returns a list of the most frequent global tags over the given period
+	 *
+	 * @param int $period Period in hours to consider posts
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function getGlobalTrendingTags(int $period)
```
Comment (nupplaphil) [inline src/Content/Widget/TrendingTags.php]: This seems for me like a `Friendica\Model\Tags` class method. I don't think we should mix it with Content methods.

File: src/Content/Feature.php [hunk 4]
```diff
@@ -84,6 +84,7 @@ public static function get($filtered = true)
 				['multi_profiles',  L10n::t('Multiple Profiles'),      L10n::t('Ability to create multiple profiles'), false, Config::get('feature_lock', 'multi_profiles', false)],
 				['photo_location',  L10n::t('Photo Location'),         L10n::t("Photo metadata is normally stripped. This extracts the location \x28if present\x29 prior to stripping metadata and links it to a map."), false, Config::get('feature_lock', 'photo_location', false)],
 				['export_calendar', L10n::t('Export Public Calendar'), L10n::t('Ability for visitors to download the public calendar'), false, Config::get('feature_lock', 'export_calendar', false)],
+				['trending_tags', L10n::t('Trending Tags'), L10n::t('Show a community page widget with a list of the most popular tags in recent public posts.'), false, Config::get('feature_lock', 'trending_tags', false)],
```
Comment (tobiasd) [inline src/Content/Feature.php]: I guess the artist who started the indentation of the `L10n` to be aligned would be happy if you could align your new entries with the rest of the list.

File: mod/community.php [hunk 5]
```diff
@@ -198,6 +198,14 @@ function community_content(App $a, $update = 0)
 		$o .= $pager->renderMinimal(count($r));
 	}
 
+	if (empty($a->page['aside'])) {
+		$a->page['aside'] = '';
+	}
+
+	if (\Friendica\Content\Feature::isEnabled(local_user(), 'trending_tags')) {
```
Comment (nupplaphil) [inline mod/community.php]: Plz create an import for both namespaces

File: src/Content/Widget/TrendingTags.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+namespace Friendica\Content\Widget;
+
+use Friendica\Core\Cache;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Database\DBA;
+use Friendica\Model\Term;
+
+class TrendingTags
```
Comment (nupplaphil) [inline src/Content/Widget/TrendingTags.php]: PhpDoc missing :)

File: src/Content/Widget/TrendingTags.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function getLocalTrendingTags(int $period)
+	{
+		$tags = Cache::get('local_trending_tags');
+
+		if (!$tags) {
+			$tagsStmt = DBA::p("SELECT t.`term`, COUNT(*) AS `score`
+FROM `term` t
```
Comment (nupplaphil) [inline src/Content/Widget/TrendingTags.php]: Wrong extends for all query lines

File: src/Content/Widget/TrendingTags.php (hunk 2 above)
Comment (annando) [reply src/Content/Widget/TrendingTags.php]: I would like to have ```Model``` classes only as an interface to the identically named tables.

Means: If it is a ```Model``` thing, then it belongs to the ```Term``` class.

Comment (nupplaphil) [reply src/Content/Widget/TrendingTags.php]: That's fine for me too :-)

As a rule of thumb for me, I say `DBA` activities are generally more bound to a `Model` class and `Renderer` activities are generally more bound to a `Content` or `Module` class.

File: src/Content/Widget/TrendingTags.php (hunk 7 above)
Comment (MrPetovan) [reply src/Content/Widget/TrendingTags.php]: You meant "indentation"?

Comment (nupplaphil) [reply src/Content/Widget/TrendingTags.php]: ... of course ... sorry ^^

**Pull request merged**